libs/ts-rest/nest/src/lib/api.decorator.ts

```diff
@@ -141,6 +162,6 @@ export const Api = (appRoute: AppRoute): MethodDecorator => {
   return applyDecorators(
     SetMetadata(tsRestAppRouteMetadataKey, appRoute),
     methodDecorator,
-    UseInterceptors(ApiRouteInterceptor)
+    UseInterceptors(new ApiRouteInterceptor(appRoute))
```

Comment:
This will create a new `ApiRouteInterceptor` instance for each endpoint, therefore increasing the memory usage of the process significantly. As the `appRoute` is already part of the request handler's metadata you can access it in the interceptor using this code on lines 52-55

```typescript
    const appRoute: AppRoute | undefined = Reflect.getMetadata(
      tsRestAppRouteMetadataKey,
      context.getHandler()
    );
```

Reply:
Nice, thanks for pointing that out @Gabrola